locked-asset/lkmex-transfer/src/lib.rs

```diff
@@ -49,14 +60,27 @@ pub trait LkmexTransfer:
         let receiver_last_transfer_mapper = self.receiver_last_transfer_epoch(&receiver);
         self.check_address_on_cooldown(&receiver_last_transfer_mapper);
         let funds = self.get_unlocked_funds(&receiver, &sender);
+        self.add_energy_to_destination(receiver.clone(), &funds);
         self.send().direct_multi(&receiver, &funds);
         self.locked_funds(&receiver, &sender).clear();
         self.all_senders(&receiver).swap_remove(&sender);
 
         let current_epoch = self.blockchain().get_block_epoch();
         receiver_last_transfer_mapper.set(current_epoch);
+    }
+
+    #[endpoint(cancelTransfer)]
+    fn cancel_transfer(&self, sender: ManagedAddress, receiver: ManagedAddress) {
+        self.require_caller_has_admin_permissions();
+        let locked_funds_mapper = self.locked_funds(&receiver, &sender);
+        require!(!locked_funds_mapper.is_empty(), TRANSFER_NON_EXISTENT);
+
+        let locked_funds = locked_funds_mapper.get();
+        locked_funds_mapper.clear();
+        self.all_senders(&receiver).swap_remove(&sender);
```

Comment:
We should probably remove the cooldown for sender as well in case of cancel:
`self.sender_last_transfer_epoch(&sender).clear();`